tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php

```diff
@@ -721,29 +717,45 @@ public function testInvalidOrientation(): void
      */
     public function testFindByAssociationArray(): void
     {
+        $address1          = new CmsAddress();
+        $address1->country = 'Germany';
+        $address1->zip     = '12345';
+        $address1->city    = 'Berlin';
+        $user1             = new CmsUser();
+        $user1->username   = 'nifnif';
+        $user1->name       = 'Nif-Nif';
+        $user1->setAddress($address1);
+
+        $address2          = new CmsAddress();
+        $address2->country = 'France';
+        $address2->zip     = '54321';
+        $address2->city    = 'Paris';
+        $user2             = new CmsUser();
+        $user2->username   = 'noufnouf';
+        $user2->name       = 'Nouf-Nouf';
+        $user2->setAddress($address2);
+
+        $address3          = new CmsAddress();
+        $address3->country = 'Spain';
+        $address3->zip     = '98765';
+        $address3->city    = 'Madrid';
+        $user3             = new CmsUser();
+        $user3->username   = 'nafnaf';
+        $user3->name       = 'Naf-Naf';
+        $user3->setAddress($address3);
+
+        $this->_em->persist($user1);
+        $this->_em->persist($user2);
+        $this->_em->persist($user3);
+        $this->_em->flush();
+
+        $ids = [$user1->id, $user2->id, $user3->id];
+        $this->_em->clear();
+
         $repo = $this->_em->getRepository(CmsAddress::class);
-        $repo->findBy(['user' => [1, 2, 3]]);
-
-        if (! class_exists(LoggingMiddleware::class)) {
-            // DBAL 2 logs queries before resolving parameter positions
-            self::assertSame(
-                [
-                    'sql' => 'SELECT t0.id AS id_1, t0.country AS country_2, t0.zip AS zip_3, t0.city AS city_4, t0.user_id AS user_id_5 FROM cms_addresses t0 WHERE t0.user_id IN (?)',
-                    'params' => [[1, 2, 3]],
-                    'types' => [Connection::PARAM_INT_ARRAY],
-                ],
-                $this->getLastLoggedQuery()
-            );
-        } else {
-            self::assertSame(
-                [
-                    'sql' => 'SELECT t0.id AS id_1, t0.country AS country_2, t0.zip AS zip_3, t0.city AS city_4, t0.user_id AS user_id_5 FROM cms_addresses t0 WHERE t0.user_id IN (?, ?, ?)',
-                    'params' => [1 => 1, 2 => 2, 3 => 3],
-                    'types' => array_fill(1, 3, ParameterType::INTEGER),
-                ],
-                $this->getLastLoggedQuery()
-            );
-        }
+        $pigs = $repo->findBy(['user' => $ids]);
+
+        self::assertCount(3, $pigs);
```

Comment:
Given that the previous version of the test makes an assertion regarding the query parameters, here, we need to check if the same three addresses have been returned, not just three arbitrary addresses.

Reply:
Ok, I'll add a wolf with a 4th address in there.